packages/components/src/tools-panel/tools-panel/hook.ts

```diff
@@ -46,20 +57,22 @@ export function useToolsPanel( props ) {
 	}, [ wasResetting ] );
 
 	// Allow panel items to register themselves.
-	const [ panelItems, setPanelItems ] = useState( [] );
+	const [ panelItems, setPanelItems ] = useState( <ToolsPanelItem[]>[] );
 
-	const registerPanelItem = ( item ) => {
+	const registerPanelItem = ( item: ToolsPanelItem ) => {
 		setPanelItems( ( items ) => [ ...items, item ] );
 	};
 
 	// Panels need to deregister on unmount to avoid orphans in menu state.
 	// This is an issue when panel items are being injected via SlotFills.
-	const deregisterPanelItem = ( label ) => {
+	const deregisterPanelItem = ( label: string ) => {
 		// When switching selections between components injecting matching
 		// controls, e.g. both panels have a "padding" control, the
 		// deregistration of the first panel doesn't occur until after the
 		// registration of the next.
-		const index = panelItems.findIndex( ( item ) => item.label === label );
+		const index = panelItems.findIndex(
+			( item: ToolsPanelItem ) => item.label === label
+		);
```

Comment:
The changes to these lines are not necessary, since TypeScript knows that `panelItems` is of type `ToolsPanelItem[]`, and that therefore, item is of the same type. The previous version of the code should still work fine:

```suggestion
		const index = panelItems.findIndex( ( item ) => item.label === label );
```